src/reactviews/pages/ConnectionDialog/components/advancedOptionsDrawer.component.tsx

```diff
@@ -35,11 +37,37 @@ export const AdvancedOptionsDrawer = ({
     setIsAdvancedDrawerOpen: React.Dispatch<React.SetStateAction<boolean>>;
 }) => {
     const context = useContext(ConnectionDialogContext);
+    const [searchSettingsText, setSearchSettingText] = useState<string>("");
+    const [userOpenedSections, setUserOpenedSections] = useState<string[]>(["General"]);
 
     if (context === undefined) {
         return undefined;
     }
 
+    const doesGroupHaveVisibleOptions = (group: ConnectionComponentGroup) => {
+        return group.options.some((optionName) => {
+            if (context.state.formComponents[optionName]?.hidden === true) {
+                return false;
+            }
+            if (searchSettingsText) {
+                return optionName.toLowerCase().includes(searchSettingsText.toLowerCase());
+            } else {
+                return true;
+            }
+        });
+    };
+
+    const isOptionVisible = (optionName: keyof IConnectionDialogProfile) => {
+        if (context.state.formComponents[optionName]?.hidden === true) {
+            return false;
+        }
+        if (searchSettingsText) {
+            return optionName.toLowerCase().includes(searchSettingsText.toLowerCase());
```

Comment:
This looks like it would only match on the connection property name, I think?

We should really be matching on both the property name (`connectRetryCount`) and the display name (`Connect retry count`).